ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/IcebergOptimizeCommit.java

```diff
@@ -214,13 +214,29 @@ private void majorCommit(ArcticTable arcticTable,
       }).filter(Objects::nonNull).collect(Collectors.toSet());
 
       // rewrite DataFiles
-      RewriteFiles rewriteFiles = baseArcticTable.newRewrite();
-      rewriteFiles.set(SnapshotSummary.SNAPSHOT_PRODUCER, CommitMetaProducer.OPTIMIZE.name());
+      RewriteFiles rewriteDataFiles = baseArcticTable.newRewrite();
       if (baseSnapshotId != TableOptimizeRuntime.INVALID_SNAPSHOT_ID) {
-        rewriteFiles.validateFromSnapshot(baseSnapshotId);
+        rewriteDataFiles.validateFromSnapshot(baseSnapshotId);
+        long sequenceNumber = arcticTable.asUnkeyedTable().snapshot(baseSnapshotId).sequenceNumber();
+        rewriteDataFiles.rewriteFiles(deleteDataFiles, addDataFiles, sequenceNumber);
```

Comment:
Hi @hzluting 
So now we use the snapshot id instead of creating a new one? 
major commit: sequence_num_1
current snapshot: sequence_num_1, but it has eq-delete files
when we commit the rewrite files, iceberg validate new deletes for these data files, then throw the Exception.
Is my understanding correct?

Reply:
The snapshot is still create a new one, but the sequence number of the DataFiles in this commit is the same as the snapshot when optimizing plan. So the eq-delete files produced by flink in optimizing process still affect the new DataFiles produced by optimizing, we don't need to check whether has new eq-delete files.
Once we commit the rewrite files with the sequence number, Iceberg won't validate eq-delete files, and we can commit success.(The detail is in MergingSnapshotProducer#validateNoNewDeletesForDataFiles())

Reply:
> 

Thanks for your comment! It makes sense